apps/emqx/src/emqx_broker.erl

```diff
@@ -159,15 +159,7 @@ do_subscribe(Topic, SubPid, SubOpts) when is_binary(Topic) ->
     %% https://emqx.atlassian.net/browse/EMQX-10214
     I = emqx_broker_helper:get_sub_shard(SubPid, Topic),
```

Comment:
Another idea, seemingly close to what @zmstone proposed: bump this counter in a pool worker process (together with inserting `?SUBSCRIBER` / `?SUBOPTIONS`) to ensure consistency, and rely on its value to make decisions on routing changes.

Pros seem to be less lookups, and load is slightly less skewed towards 0-th shard (though this one is almost negligible). Cons are that (supposedly?) insert to `?SUBOPTION` has to be performed in the pool worker process, and potentially unfair client-to-shard allocation, since client process can only read this value to pick a shard but not bump it. Curious what do you think about it?

Reply:
This is interesting, I also tend to to think that during abrupt load spikes it may lead to slightly worse sharding.. 
At the same time, it may be quite negligible.

Reply:
Disregard that: it's still possible to have operations reordered, because there would be different processes acting on the same routes.